gftools no longer supports spaces in working directory path

Minor issue of course, seemingly I'm the only person that bothers to make nice looking folder names lol

Updating a project I'm working on from v0.9.36 to v0.9.52

```
$ pwd
/home/atk/Fonts/Neat Font Name with Spaces

$ venv/bin/gftools builder sources/config.yaml
[1/3] buildVariable
FAILED: /tmp/tmp_leo9if2
/home/atk/Fonts/Neat Font Name with Spaces/venv/bin/python3 -m gftools.builder.jobrunner fontmake --output-path /tmp/tmp_leo9if2 -o variable -m NeatFont.designspace --filter ...  --filter FlattenComponentsFilter --filter DecomposeTransformedComponentsFilter --keep-overlaps
/bin/sh: line 1: /home/atk/Fonts/Neat: No such file or directory
ninja: build stopped: subcommand failed.
```

If I get some more time I'll try to track down where this is, but for now I'm filing this issue so I don't forget :)

--- a/Lib/gftools/builder/operations/__init__.py
+++ b/Lib/gftools/builder/operations/__init__.py
@@ -10,6 +10,7 @@ from os.path import dirname
 from tempfile import NamedTemporaryFile
 
 from gftools.builder.file import File
+from gftools.utils import shell_quote
 
 
 @dataclass
@@ -52,7 +53,7 @@ class OperationBase:
             cmd = cls.rule + " $stamp"
         writer.rule(
             name,
-            f'"{sys.executable}" -m gftools.builder.jobrunner {cmd}',
+            f"{shell_quote(sys.executable)} -m gftools.builder.jobrunner {cmd}",
             description=name,
         )
         writer.newline()

--- a/Lib/gftools/builder/operations/__init__.py
+++ b/Lib/gftools/builder/operations/__init__.py
@@ -52,7 +52,7 @@ class OperationBase:
             cmd = cls.rule + " $stamp"
         writer.rule(
             name,
-            sys.executable + " -m gftools.builder.jobrunner " + cmd,
+            f'"{sys.executable}" -m gftools.builder.jobrunner {cmd}',
             description=name,
         )
         writer.newline()

--- a/Lib/gftools/utils.py
+++ b/Lib/gftools/utils.py
@@ -14,6 +14,7 @@
 # limitations under the License.
 #
 from __future__ import annotations
+from typing import Union
 import requests
 from io import BytesIO
 from zipfile import ZipFile
@@ -31,6 +32,8 @@ from google.protobuf import text_format
 import json
 from PIL import Image
 import re
+import shlex
+import subprocess
 from fontTools import unicodedata as ftunicodedata
 from fontTools.ttLib import TTFont
 from ufo2ft.util import classifyGlyphs
@@ -601,3 +604,15 @@ def open_ufo(path):
     else:  # Maybe a .ufoz
         return ufoLib2.Font.open(path)
     return False
+
+# https://github.com/googlefonts/nanoemoji/blob/fb4b0b3e10f7197e7fe33c4ae6949841e4440397/src/nanoemoji/util.py#L167-L176
+def shell_quote(s: Union[str, Path]) -> str:
+    """Quote a string or pathlib.Path for use in a shell command."""
+    s = str(s)
+    # shlex.quote() is POSIX-only, for Windows we use subprocess.list2cmdline()
+    # which converts a list of args to a command line string following the
+    # the MS C runtime rules.
+    if sys.platform.startswith("win"):
+        return subprocess.list2cmdline([s])
+    else:
+        return shlex.quote(s)

